user-interface/0001-WALL-wallet.md

```diff
@@ -0,0 +1,186 @@
+
+... so the version of the wallet app I am using works with the network I am using
+
+## Log in to a wallet
+When using a given wallet, I...
+
+- **must** select a wallet and enter the pasphrase only once per "session" (<a name="0001-WALL-016" href="#0001-WALL-016">0001-WALL-016</a>)
+
+... so that other users of my machine can not use my wallet, and I am not asked to re-enter frequently
+
+## Connecting to Dapps
+When a dapp requests use of a wallet, I...
+
+- **must** be prompted to either select a wallet or dismiss the prompt  (<a name="0001-WALL-017" href="#0001-WALL-017">0001-WALL-017</a>)
+- `TODO` am required to select what keys of a wallet to grant access to <!--(<a name="0001-WALL-018" href="#0001-WALL-018">0001-WALL-018</a>)-->
+  - `TODO:` **must** be able to select whole wallet (so that new keys are automatically shared)  <!--(<a name="0001-WALL-019" href="#0001-WALL-019">0001-WALL-019</a>)-->
+  - `TODO:` **must** be able to select specific [keys](7001-DATA-data_display.md#public-keys)<!-- (<a name="0001-WALL-020" href="#0001-WALL-020">0001-WALL-020</a>) -->
+  - `TODO` tainted keys **must** be shown as tainted (<a name="0001-WALL-021" href="#0001-WALL-021">0001-WALL-021</a>)
+  - `TODO` tainted keys **must** not be selectable (<a name="0001-WALL-073" href="#0001-WALL-073">0001-WALL-073</a>)**
+- **must** enter wallet passphrase before wallet details are shared (assuming a password has not recently been entered)(<a name="0001-WALL-022" href="#0001-WALL-022">0001-WALL-022</a>)
+- **must** be able to retrospectively revoke Dapp's access to a Wallet (<a name="0001-WALL-023" href="#0001-WALL-023">0001-WALL-023</a>)
+- **must** be able to retrospectively revoke Dapp's access particular keys in a wallet (<a name="0001-WALL-072" href="#0001-WALL-072">0001-WALL-072</a>)
+
+... so that I can control what public keys are shared with a dapp and what dapps can prompt me to sign transactions 
+
+## Approving transactions
+When a dapp sends a transaction to the wallet for signing and broadcast, I...
+
+- **must** be prompted to approve, reject or ignore the transaction (if auto approve is not on) (<a name="0001-WALL-024" href="#0001-WALL-024">0001-WALL-024</a>)
+- **must** see the details of the transaction. See [details of transaction](#transaction-detail). (<a name="0001-WALL-025" href="#0001-WALL-025">0001-WALL-025</a>)
+- **must** see any ignored/dismissed in a transactions area along with pending, approved and rejected transactions (e.g. history). See [Transactions](#transactions).
+
+... so I can verify that the transaction being sent is the one I want
+
+## View keys balances + positions/accounts
+When looking for a specific balance or asset on a given wallet and network, I...
+
+- **must** see total balances of an asset for each key (<a name="0001-WALL-027" href="#0001-WALL-027">0001-WALL-027</a>)
```

Comment:
Worth verifying if this is a must. I know some would prefer to not have any fetching accounts at all. Or at the very least the initial startup dialog should warn users or prompt them to opt in to having their public keys used to fetch data

Reply:
I added 0001-WALL-078 after the discussion but will re-phrase this to mention the ability to opt in/out of queries.
```suggestion
- **must** see total balances of an asset for each key (subject to that key included in data node queries, see 078) (<a name="0001-WALL-027" href="#0001-WALL-027">0001-WALL-027</a>)
```

Reply:
I added 0001-WALL-078 after the discussion but will re-phrase this to mention the ability to opt in/out of queries.
```suggestion
- **must** see total balances of an asset for each key (subject to that key being included in data queries (see 078)) (<a name="0001-WALL-027" href="#0001-WALL-027">0001-WALL-027</a>)
```